internal/controller/eventing/nats_test.go

```diff
@@ -0,0 +1,216 @@
+			},
+			givenManagerFactoryMock: func(_ *ecsubmanagermocks.Manager) *subscriptionmanagermocks.ManagerFactory {
+				return nil
+			},
+		},
+		{
+			name:                         "should init and start NATSSubManager",
+			givenIsNATSSubManagerStarted: false,
+			givenNATSSubManagerMock: func() *ecsubmanagermocks.Manager {
+				jetStreamSubManagerMock := new(ecsubmanagermocks.Manager)
+				jetStreamSubManagerMock.On("Init", mock.Anything).Return(nil).Once()
+				jetStreamSubManagerMock.On("Start", mock.Anything, mock.Anything).Return(nil).Once()
+				return jetStreamSubManagerMock
+			},
+			givenEventingManagerMock: func() *managermocks.Manager {
+				emMock := new(managermocks.Manager)
+				emMock.On("GetBackendConfig").Return(givenBackendConfig)
+				emMock.On("GetNATSConfig").Return(givenNATSConfig)
+				return emMock
+			},
+			givenManagerFactoryMock: func(subManager *ecsubmanagermocks.Manager) *subscriptionmanagermocks.ManagerFactory {
+				subManagerFactoryMock := new(subscriptionmanagermocks.ManagerFactory)
+				subManagerFactoryMock.On("NewJetStreamManager", mock.Anything, mock.Anything).Return(subManager).Once()
+				return subManagerFactoryMock
+			},
+			wantAssertCheck: true,
+		},
+	}
+
+	// run test cases
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			// given
+			testEnv := NewMockedUnitTestEnvironment(t)
+			logger := testEnv.Reconciler.logger.WithContext().Named(ControllerName)
+
+			// get mocks from test-case.
+			givenNATSSubManagerMock := tc.givenNATSSubManagerMock()
+			givenManagerFactoryMock := tc.givenManagerFactoryMock(givenNATSSubManagerMock)
+			givenEventingManagerMock := tc.givenEventingManagerMock()
+
+			// connect mocks with reconciler.
+			testEnv.Reconciler.isNATSSubManagerStarted = tc.givenIsNATSSubManagerStarted
+			testEnv.Reconciler.eventingManager = givenEventingManagerMock
+			testEnv.Reconciler.subManagerFactory = givenManagerFactoryMock
+			testEnv.Reconciler.NATSSubManager = nil
+			if givenManagerFactoryMock == nil {
+				testEnv.Reconciler.NATSSubManager = givenNATSSubManagerMock
+			}
+
+			// when
+			err := testEnv.Reconciler.reconcileNATSSubManager(givenEventing, logger)
+
+			// then
+			require.NoError(t, err)
+			require.NotNil(t, testEnv.Reconciler.NATSSubManager)
+			require.True(t, testEnv.Reconciler.isNATSSubManagerStarted)
+			if tc.wantAssertCheck {
+				givenNATSSubManagerMock.AssertExpectations(t)
+				givenManagerFactoryMock.AssertExpectations(t)
+				givenEventingManagerMock.AssertExpectations(t)
+			}
+		})
+	}
+}
+
+func Test_stopNATSSubManager(t *testing.T) {
+	t.Parallel()
+
+	// define test cases
+	testCases := []struct {
+		name                         string
+		givenNATSSubManagerMock      func() *ecsubmanagermocks.Manager
+		givenIsNATSSubManagerStarted bool
+		wantError                    error
+		wantAssertCheck              bool
+	}{
+		{
+			name: "should do nothing if NATSSubManager is not initialised",
```

Comment:
Can the NATSSubManager be initialised but not started?

Reply:
yes, if it was initialised but then it fails to start.

Reply:
https://github.com/kyma-project/eventing-manager/pull/50